ffi/derive/src/convert.rs

```diff
@@ -197,16 +205,24 @@ fn derive_try_from_repr_c_for_opaque_item(name: &Ident) -> TokenStream2 {
     }
 }
 
-fn derive_try_from_repr_c_for_opaque_item_slice(name: &Ident) -> TokenStream2 {
+fn derive_try_from_repr_c_for_opaque_item_slice(input: &DeriveInput) -> TokenStream2 {
+    let name = &input.ident;
+    let mut generics = input.generics.clone();
+    let lifetime: syn::Lifetime = syn::parse_quote!('slice);
+    let lifetimes = &[lifetime.clone()];
+    let trait_bounds = &[syn::parse_quote!(core::clone::Clone)];
```

Comment:
Required because of cloning inside this method.

I was surprised to find out that rust accept smt like `T: Clone + Clone` [just fine](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b7455998d82a3e368db1c731831f8776).

So we shouldn't worry about this.

My another concern is lifetime name collisions so maybe we should give our lifetimes more obfuscated name.

Reply:
> My another concern is lifetime name collisions so maybe we should give our lifetimes more obfuscated name.

Definitely. Something like `__iroha_ffi_itm`